corgi/core/models.py

```diff
@@ -1575,15 +1575,21 @@ def is_srpm(self):
         return self.type == Component.Type.RPM and self.arch == "src"
 
     def get_nvr(self) -> str:
+        # Many GOLANG components don't have a version or release set
+        # so don't return an oddly formatted NVR, like f"{name}-"
+        version = f"-{self.version}" if self.version else ""
         release = f"-{self.release}" if self.release else ""
-        return f"{self.name}-{self.version}{release}"
+        return f"{self.name}{version}{release}"
 
     def get_nevra(self) -> str:
         epoch = f":{self.epoch}" if self.epoch else ""
+        # Many GOLANG components don't have a version or release set
+        # so don't return an oddly formatted NEVRA, like f"{name}-.{arch}"
+        version = f"-{self.version}" if self.version else ""
         release = f"-{self.release}" if self.release else ""
         arch = f".{self.arch}" if self.arch else ""
```

Comment:
Perhaps we could add an additional check here for namespace = REDHAT, and type in ["RPM","CONTAINER_IMAGE"] and only append the arch in that case?